bgpd/bgp_bmp.c

```diff
@@ -548,25 +847,65 @@ static int bmp_send_peerup(struct bmp *bmp)
 	/* Walk down all peers */
 	for (ALL_LIST_ELEMENTS_RO(bmp->targets->bgp->peer, node, peer)) {
 		s = bmp_peerstate(peer, false);
+		if (!s)
+			continue;
+
 		pullwr_write_stream(bmp->pullwr, s);
 		stream_free(s);
 	}
 
 	return 0;
 }
 
+static int bmp_send_peerup_vrf(struct bmp *bmp)
+{
+
+	struct bmp_bgp *bmpbgp = bmp->targets->bmpbgp;
+
+	/* send unconditionally because state may have been set before the
+	 * session was up. and in this case the peer up has not been sent.
+	 */
+	bmp_bgp_update_vrf_status(bmpbgp, vrf_state_unknown);
+
+	zlog_info("bmp: sending peer state message up=%d for session %s (%s)", bmpbgp->vrf_up == vrf_state_up, bmpbgp->bgp->name, bmpbgp->bgp->name_pretty);
+	struct stream *s = bmp_peerstate(bmpbgp->bgp->peer_self, bmpbgp->vrf_up == vrf_state_down);
+
+	if (!s) {
+		zlog_warn("bmp: peer state message error");
```

Comment:
i can change to something like `zlog_warn("bmp: vrf peer state failed, skipping message.");`
if an error occurs this is important for bmp to tell us that we skipped a bmp peer up/down message for the vrf since some collectors will not work without it.